iFixit/react-commerce
# Refactor Shopify Storefront SDK codegen

This PR has been extracted from the work I've done on the RSC demo repo

## Changelog

- **Simplifies the Storefront SDK code generation**: Use Shopify [mock.shop](https://mock.shop/) as the source of truth for the api schema. This has two advantages:
  -  We no longer need to have test store secrets env and a script to download the graphql schema
  - The schema will always be in sync with the version we're using in production
- **Use one Storefront version**: Currently we're using two different version of the API (one for the cart and one for the product page). This makes sure we're using only one version, the one we've set in the env file. We're also declaring `NEXT_PUBLIC_SHOPIFY_STOREFRONT_VERSION` in `.env` which is shared by all environments, as there's no reason to differentiate between dev and production (and it's risky)
- **Update Storefront version** : This brings the version we're using to the latest stable release `2023-10`

## QA

1. Open Vercel preview for a [product page](https://react-commerce-git-refactor-storefront-codegen-ifixit.vercel.app/products/pro-tech-toolkit)
2. Verify that the product page still renders product data from Shopify correctly
3. Add something to the cart, try to checkout and verify that you are redirected to the Shopify checkout
